src/Core/SortCursor.h

```diff
@@ -441,21 +514,65 @@ class SortingHeap
 
             child_it = begin + child_idx;
 
-            if ((child_idx + 1) < size && *child_it < *(child_it + 1))
+            if ((child_idx + 1) < size && (*child_it).greater(*(child_it + 1)))
```

Comment:
I'm not sure that's a problem, but before we had `operator <` (smaller) and now it's `greater`. Shouldn't you swap arguments?

Reply:
It was confusing so I removed it. `operator <` called function `greater`.